crates/oxc_linter/src/lib.rs

```diff
@@ -49,16 +56,23 @@ impl Linter {
     }
 
     #[must_use]
-    pub fn run(&self, semantic: &Rc<Semantic>) -> Vec<Error> {
-        let ctx = LintContext::new(semantic.clone());
+    pub fn run(&self, semantic: &Rc<Semantic>, source_text: &str) -> LintRunResult {
+        let mut ctx = LintContext::new(semantic.clone());
```

Comment:
Did you consider using `RefCell` for `fixes` such that we don need to specified that the ctx is `mut`.

Reply:
Just like `diagnostics`

Reply:
Yes that's something I'm going to update when I move to mutating `ctx` directly inside `Rule::run` 👍🏼